webdriver/tests/bidi/browsing_context/activate/activate.py

```diff
@@ -0,0 +1,52 @@
+
+import pytest
+
+from webdriver.bidi.modules.script import ContextTarget
+from . import get_visibility_state, is_selector_focused
+
+@pytest.mark.asyncio
+async def test_activate(bidi_session, new_tab):
+
+    assert await get_visibility_state(bidi_session, new_tab) == 'hidden'
+
+    await bidi_session.browsing_context.activate(context=new_tab["context"])
+
+    assert await get_visibility_state(bidi_session, new_tab) == 'visible'
+
+
+@pytest.mark.asyncio
+async def test_deactivates_other_context(bidi_session, new_tab, top_context):
+    await bidi_session.browsing_context.activate(context=top_context["context"])
+
+    assert await get_visibility_state(bidi_session, top_context) == 'visible'
+    assert await get_visibility_state(bidi_session, new_tab) == 'hidden'
+
+    await bidi_session.browsing_context.activate(context=new_tab["context"])
+
+    assert await get_visibility_state(bidi_session, top_context) == 'hidden'
+    assert await get_visibility_state(bidi_session, new_tab) == 'visible'
+
+
+@pytest.mark.asyncio
+async def test_keeps_focused_area(bidi_session, inline, new_tab, top_context):
+    await bidi_session.browsing_context.activate(context=new_tab["context"])
+    assert await get_visibility_state(bidi_session, new_tab) == 'visible'
+
+    await bidi_session.browsing_context.navigate(context=new_tab["context"],
+                                                 url=inline("<input>"),
+                                                 wait="complete")
+
+    await bidi_session.script.evaluate(
+        expression="""document.querySelector("input").focus()""",
```

Comment:
Suggestion: in addition to this, we could also test `<input autofocus>` (with no script evaluation)